src/transactions/InvokeHostFunctionOpFrame.cpp

```diff
@@ -440,6 +441,20 @@ InvokeHostFunctionOpFrame::doApply(Application& app, AbstractLedgerTxn& ltx,
 
             metrics.noteReadEntry(isCodeKey(lk), keySize, entrySize);
 
+            if (isSorobanEntry(lk) &&
```

Comment:
You should reuse `validateContractLedgerEntry` (you're missing the contract code case here). I also think we need to move it to a separate utility and use in all the operations.